gcc/config/riscv/riscv.c

```diff
@@ -3609,12 +3630,7 @@ riscv_hard_regno_mode_ok_p (unsigned int regno, enum machine_mode mode)
     }
 
   if (FP_REG_P (regno))
-    {
-      if (mclass == MODE_FLOAT
-	  || mclass == MODE_COMPLEX_FLOAT
-	  || mclass == MODE_VECTOR_FLOAT)
-	return size <= UNITS_PER_HWFPVALUE;
-    }
+    return true;
```

Comment:
That makes sense to me in theory, but I'm concerned about how it interacts with recoding of sub-word floating-point values.  If you spill a 32-bit type using FSD (or FMV.X.D), then the only guarantee is that reloading it with FLD (or FMV.D.X) will re-create the original value.  The value that gets spilled to memory (or an integer register) is implementation-defined.  For example, after executing the code

    fmv.s.x ft0, t0
    fmv.x.d t1, ft0
    fmv.d.x ft1, t1
    fmv.x.s t2, ft1

it is guaranteed that ft0 == ft1, and it is guaranteed that t0 == t2, but it is _not_ guaranteed that t0 == t1.

I think that CANNOT_CHANGE_MODE_CLASS handles this already, but I want to be certain this change won't introduce bugs in certain implementations.